python/composio/tools/toolset.py

```diff
@@ -435,6 +435,7 @@ def check_connected_account(self, action: ActionType) -> None:
             # Normalize app names/ids coming from API
             connection.appUniqueId.upper()
             for connection in self._connected_accounts
+            if connection.entityId == entity_id
         ]:
             raise ComposioSDKError(
                 f"No connected account found for app `{action.app}`; "
```

Comment:
Consider enhancing the error message to include the entity ID for better debugging:
```python
f"No connected account found for app `{action.app}` on entity `{entity_id}`; Run `composio add {action.app.lower()}` to fix this"
```